src/utils/WebUtils.ts

```diff
@@ -19,8 +23,23 @@ const loadScript = async (scriptUrl: string, callbackOnload: CallableFunction =
     });
 
     scriptElement.addEventListener('error', async () => {
-      await callbackError();
-      reject();
+      if (++attempt >= retries) {
+        await callbackError();
+
+        // Reference: https://developer.mozilla.org/en-US/docs/Web/API/Element/error_event
+        return reject(new Error("Resource failed to load, or can't be used."));
+      }
+
+      scriptElement.remove();
+
+      const exponentialBackoffWaitTime = Math.min((2 ** attempt) + Math.random(), maxBackoffSeconds) * 1000;
+      await sleep(exponentialBackoffWaitTime);
+
+      try {
+        await loadScript(scriptUrl, callbackOnload, callbackError, retries, attempt);
+      } catch (e) {
+        reject(e);
```

Comment:
Do you think, we need to add more telemetry here or is it sufficient while making a call ?

Reply:
It's good enough. It's already the best conclusion we can find with this type of error.